tests/kernel/interrupt/src/interrupt_offload.c

```diff
@@ -0,0 +1,266 @@
+/*
+ * Copyright (c) 2021 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <ztest.h>
+#include <irq_offload.h>
+#include <interrupt_util.h>
+
+#define STACK_SIZE	1024
+#define NUM_WORK	4
+
+static struct k_work offload_work[NUM_WORK];
+static struct k_work_q wq_queue;
+static K_THREAD_STACK_DEFINE(wq_stack, STACK_SIZE);
+static K_THREAD_STACK_DEFINE(tstack, STACK_SIZE);
+static struct k_thread tdata;
+
+static struct k_sem start_sem;
+static struct k_sem end_sem;
+
+/*
+ * This global variable control if the priority of offload job
+ * is greater than the original thread.
+ */
+static bool offload_job_prio_higher;
+
+static volatile int orig_t_keep_run;
+
+enum {
+	TEST_OFFLOAD_MULTI_JOBS,
+	TEST_OFFLOAD_IDENTICAL_JOBS
+};
+
+struct interrupt_param {
+	struct k_work *work;
+};
+
+static struct interrupt_param irq_param;
+
+static void entry_offload_job(struct k_work *work)
+{
+	printk("doing offload job\n");
```

Comment:
Please use TC_PRINT instead of printk in tests.

Reply:
I think some prints mostly are for the developer purposes. Some of them you can just replace with C comments /* doing offload job */
I don't think we have to make dirty the output of the test.

Reply:
Hi @maksimmasalski , I have addressed these two suggestions in my code, could you please take a look at it again, thanks!